### PR TITLE
refactor(app): Add new ConnectionCard with interface and WiFi details

### LINKED ISSUE #2346: Manage Robot Connection: Updated Robot Connection Card
As a Run App user, I would like to be able to see full details of my robot's connection state, so that I can make informed decision when managing my robot's connection.

## Acceptance criteria
- Update 'Robot Connection' card information as follows:
  - [ ] Display active connection method
    - "Your app is currently connected to your robot via [ Wifi | USB ]
  - [ ] Display the robot's internet connection status (see copy below)
  - [ ] Display USB connection details
    - [ ] Wired IP Address
    - [ ] Wired MAC Address 
  - [ ] Display WiFi connection details and management options
    - [ ] Wireless IP Address
    - [ ] Wireless MAC address
    - Dropdown filed to select/add WiFi connections (functionality not covered in this user story)

## Design
This design still requires discussion/refinement:
- https://zpl.io/bAExGmo

### Copy (Internet Connection Status)
Internet connection status may be one of the following:
- `"status": "none"` > "The robot is not connected to any network."
- `"status": "portal"` > "The robot is behind a captive portal and cannot reach the full Internet."
- `"status": "limited"` > "The  robot is connected to a network, but it has no access to the Internet."
- `"status": "full"` > "The robot is connected to a network and has full access to the Internet."
- `"status": "unknown"` > "Unknown"

### LINKED ISSUE #2347: Manage Robot Connection: Display Wireless Network Details in Dropdown
As a Run App user, I would like to see wifi strength and security details for wifi networks in the networks dropdown, so I can make an informed network choice.

## Acceptance Criteria
- Display 'lock' icon next to wifi network name if network requires authentication
- Display 'wifi' icon with network strength next to network name
- Display 'checkbox' icon next to actively connected network, if any
- Sort by connectivity (active connection at top), then alphabetically a-z.

## Design
- https://zpl.io/Vkn9mG5

### PR DESCRIPTION
**Pull request by mcous**
## overview

This PR closes #2346 and closes #2347 by adding a new connection card to the robot settings page that displays the robot's:

- Internet connectivity status
- Current connection to the app (USB or Wi-Fi)
- IP and MAC address of the Wi-Fi interface
- IP and MAC address of the Ethernet interface
- Available Wi-Fi networks with each network item displaying:
    - A checkmark if the network is currently active
    - A lock if the network is secured
    - A signal strength indicator

Note: Wi-Fi select list will require a styling pass which we will add on as a new ticket.

## changelog

- refactor(app): Add new ConnectionCard with interface and WiFi detail
    - Note: refactor because behind feature flag; flipping the flag will be the `feat` commit

## review requests

New connectivity stuff is behind a (too long) feature flag, so make sure you launch the app with that flag enabled:

```shell
# via make dev
# install new dependencies
make install
# run the app with the flag enabled
make -C app dev OT_APP_DEV_INTERNAL__MANAGE_ROBOT_CONNECTION__NEW_CARD=1

# via a built artifact
./path/to/Opentrons --devInternal.manageRobotConnection.newCard
```

(We're adding internal dev feature flags to the `devInternal` config sub object to avoid having to modify the defaults that could potential collide with future user-facing flags)

### component library

We need to check that the change we made to `Card` css doesn't mess with PD. We needed to change to `overflow: visible` to ensure the Wi-Fi dropdown (which is no longer a native `<select>` actually renders outside the card.

### CODE REVIEW

File: components/src/structure/structure.css [hunk 1]
```diff
@@ -79,15 +79,16 @@
 /* Card and card content styles */
 .card {
   position: relative;
-  overflow: auto;
+  overflow: visibile;
```
Comment (IanLondon) [inline components/src/structure/structure.css]: `visibile`?

File: app/package.json [hunk 2]
```diff
@@ -38,6 +38,7 @@
     "react-router": "^4.2.0",
     "react-router-dom": "^4.2.2",
     "react-router-redux": "^5.0.0-alpha.6",
+    "react-select": "^2.1.0",
```
Comment (IanLondon) [inline app/package.json]: this looks very useful, maybe we can solve some PD design problems we've been punting on with this! @b-cooper

File: app/src/components/RobotSettings/connection.js [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+type AvailableNetworksProps = {list: ?WifiNetworkList}
+
+export function AvailableNetworks (props: AvailableNetworksProps) {
+  const list = props.list || []
+  const options = list.map(NetworkOption)
+  return (
+    <Select
+      value={options[0]}
+      className={styles.wifi_dropdown}
+      onChange={e => console.log(e)}
```
Comment (IanLondon) [inline app/src/components/RobotSettings/connection.js]: this may be a lingering `log`

Comment (mcous) [reply app/src/components/RobotSettings/connection.js]: Oops, should've flagged as a TODO. Current;y hanging out as a debugging tool until we get to connect functionality

**Pull request merged**